packages/p2p/src/components/app.jsx

```diff
@@ -45,13 +63,47 @@ class App extends Component {
         this.setState({ parameters: null });
     };
 
-    setIsAgent = async () => {
-        const agent_info = await requestWS({ p2p_agent_info: 1 });
+    setIsAgent = () =>
+        new Promise(async resolve => {
+            const agent_info = await requestWS({ p2p_agent_info: 1 });
 
-        /* if there is no error means its an agent else its a client */
-        if (!agent_info.error) {
-            this.setState({ agent_id: agent_info.p2p_agent_info.agent_id, is_agent: true });
-        }
+            /* if there is no error means its an agent else its a client */
+            if (!agent_info.error) {
+                await this.setState({ agent_id: agent_info.p2p_agent_info.agent_id, is_agent: true });
+                resolve();
+            }
+            resolve();
+        });
+
+    handleNotifications = updated_orders => {
+        let notifications = 0;
+
+        updated_orders.forEach(order => {
+            const modified_order = new OrderInfo(order);
+
+            const user_state = {
+                is_agent_buyer: this.state.is_agent && modified_order.is_buyer,
+                is_agent_seller: this.state.is_agent && !modified_order.is_buyer,
+                is_client_buyer: !this.state.is_agent && modified_order.is_buyer,
+                is_client_seller: !this.state.is_agent && !modified_order.is_buyer,
+            };
+
+            const order_state = {
+                pending: modified_order.is_pending,
+                'buyer-confirmed': modified_order.is_buyer_confirmed,
+            };
+
+            Object.keys(notifications_map).forEach(notif_state => {
+                if (order_state[notif_state]) {
+                    Object.keys(notifications_map[notif_state]).forEach(notif_condition => {
```

Comment:
why are we having three nested loops? we are already looping each order, we just need to check if this current order has a status matching to the current user state then add one to notifications

Reply:
yes, thats if we use if else, but, since we want to map it into our defined object, we need to loop through the defined map as well

Reply:
let's change it to the if-else :D too many loops now :D

Reply:
exactly 😅